train.lua

```diff
@@ -32,349 +35,195 @@ cmd:text('Train a character-level language model')
 cmd:text()
 cmd:text('Options')
 -- data
-cmd:option('-data','tinyshakespeare','name of data directory. Should contain the file input.txt with input data')
-cmd:option('-back','cuda','cpu|cuda|cl')
-cmd:option('-len', -1, 'only use this many characters from input data. -1 for all data')
```

Comment:
we need the `len` option for doing unit-testing, and sanity testing.  Well.. we could make other data sets I suppose, is another way.  But I think it's fairly standard to be able to load a subset of the training data?